WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Grouped Products/ProductListMultiSelectorSearchUICommand.swift

```diff
@@ -0,0 +1,123 @@
+import Yosemite
+
+/// Implementation of `SearchUICommand` for selecting linked products to a grouped product from search UI.
+final class ProductListMultiSelectorSearchUICommand: SearchUICommand {
+    typealias Model = Product
+    typealias CellViewModel = ProductsTabProductViewModel
+    typealias ResultsControllerModel = StorageProduct
+
+    let searchBarPlaceholder = NSLocalizedString("Search all products", comment: "Products Search Placeholder")
+
+    let searchBarAccessibilityIdentifier = "product-search-screen-search-field"
+
+    let cancelButtonAccessibilityIdentifier = "product-search-screen-cancel-button"
+
+    private let siteID: Int64
+
+    private let excludedProductIDs: [Int64]
+    private var selectedProductIDs: [Int64] = []
+
+    typealias Completion = (_ selectedProductIDs: [Int64]) -> Void
+    private let onCompletion: Completion
+
+    init(siteID: Int64, excludedProductIDs: [Int64], onCompletion: @escaping Completion) {
+        self.siteID = siteID
+        self.excludedProductIDs = excludedProductIDs
+        self.onCompletion = onCompletion
+    }
+
+    func createResultsController() -> ResultsController<ResultsControllerModel> {
+        let storageManager = ServiceLocator.storageManager
+        let predicate = NSPredicate(format: "siteID == %lld AND NOT(productID in %@)", siteID, excludedProductIDs)
```

Comment:
yeaa I always felt like making a mistake when writing an `NSPredicate`. I added a test case in [`ProductListMultiSelectorSearchUICommandTests.testCommandCreatesResultsControllerExcludingSpecifiedProductIDs`](https://github.com/woocommerce/woocommerce-ios/blob/977a21a5b6eb42bbe13e76d6aad55af19b67774e/WooCommerce/WooCommerceTests/ViewRelated/Search/Product/ProductListMultiSelectorSearchUICommandTests.swift#L18-L40), lemme know if you think it could have tests!